server/src/main/java/org/opensearch/common/cache/store/config/CacheConfig.java

```diff
@@ -44,6 +45,10 @@ public class CacheConfig<K, V> {
 
     private final RemovalListener<K, V> removalListener;
 
+    // Serializers for keys and values. Not required for all caches.
+    private final Serializer<K, ?> keySerializer;
```

Comment:
Should we have some contract here to specify that key or value should serializable?
something like ```? extends Serializable``` ? Possible?

Reply:
Hmm I'm not sure we could use Java's Serializable, as the cache implementation might not care about that interface (ehcache doesn't for example). Maybe it'd be better to just enforce it being <K, byte[]>? That seems like a pretty "neutral" thing to serialize to, and cache implementations can handle it as needed